core/build_playerglobal/src/lib.rs

```diff
@@ -145,55 +215,50 @@ fn write_native_table(data: &[u8], out_dir: &Path) -> Result<(), Box<dyn std::er
+        let class_id = if let TraitKind::Class { slot_id, .. } = trait_.kind {
+            slot_id
+        } else {
+            return;
+        };
 
-        let trait_name = &abc.constant_pool.multinames[trait_.name.0 as usize - 1];
+        let class_name_idx = abc.instances[class_id as usize - 1].name.0;
+        let class_name = resolve_multiname_name(
+            &abc,
+            &abc.constant_pool.multinames[class_name_idx as usize - 1],
+        );
 
-        if let Some(parent) = parent {
-            // This is a method defined inside the class. Append the class namespace
-            // (the package) and the class name.
-            // For example, a namespace of "flash.system" and a name of "Security"
-            // turns into the path "flash::system::security"
-            let multiname = &abc.constant_pool.multinames[parent.0 as usize - 1];
-            path += &flash_to_rust_path(resolve_multiname_ns(&abc, multiname));
-            path += "::";
-            path += &flash_to_rust_path(resolve_multiname_name(&abc, multiname));
-            path += "::";
-        } else {
-            // This is a freestanding function. Append its namespace (the package).
-            // For example, the freestanding function "flash.utils.getDefinitionByName"
-            // has a namespace of "flash.utils", which turns into the path
-            // "flash::utils"
-            path += &flash_to_rust_path(resolve_multiname_ns(&abc, trait_name));
-            path += "::";
-        }
+        let method_name = "::".to_string() + &flash_to_rust_path(class_name) + "_allocator";
 
-        // Append the trait name - this corresponds to the actual method
-        // name (e.g. `getDefinitionByName`)
-        path += method_prefix;
-        path += &flash_to_rust_path(resolve_multiname_name(&abc, trait_name));
+        for metadata_idx in &trait_.metadata {
+            let metadata = &abc.metadata[metadata_idx.0 as usize];
+            let name = &abc.constant_pool.strings[metadata.name.0 as usize - 1];
+            match name.as_str() {
+                RUFFLE_METADATA_NAME => {}
+                _ => panic!("Unexpected class metadata {:?}", name),
+            }
 
-        // Now that we've built up the path, convert it into a `TokenStream`.
-        // This gives us something like
-        // `crate::avm2::globals::flash::system::Security::allowDomain`
-        //
-        // The resulting `TokenStream` is suitable for usage with `quote!` to
-        // generate a reference to the function pointer that should exist
-        // at that path in Rust code.
-        let path_tokens = TokenStream::from_str(&path).unwrap();
-        rust_paths[method_id.0 as usize] = quote! { Some(#path_tokens) };
+            for item in &metadata.items {
+                let key = &abc.constant_pool.strings[item.key.0 as usize - 1];
+                let value = &abc.constant_pool.strings[item.value.0 as usize - 1];
```

Comment:
Should handle the case where key us `0` when there's a keyless metadata like so:
`[Foo(Bar)]`